pkg/watch/manager.go

```diff
@@ -424,22 +424,34 @@ func (r *recordKeeper) Remove(rm map[string][]schema.GroupVersionKind) {
 	}
 }
 
-func (r *recordKeeper) Get() map[string]map[schema.GroupVersionKind]Vitals {
+func (r *recordKeeper) Get() map[string]map[schema.GroupVersionKind]vitals {
 	r.intentMux.RLock()
 	defer r.intentMux.RUnlock()
-	cpy := make(map[string]map[schema.GroupVersionKind]Vitals)
+	cpy := make(map[string]map[schema.GroupVersionKind]vitals)
 	for k := range r.intent {
-		cpy[k] = make(map[schema.GroupVersionKind]Vitals)
+		cpy[k] = make(map[schema.GroupVersionKind]vitals)
 		for k2, v := range r.intent[k] {
 			cpy[k][k2] = v
 		}
 	}
 	return cpy
 }
 
+func (r *recordKeeper) GetGVK() []schema.GroupVersionKind {
+	r.intentMux.RLock()
+	defer r.intentMux.RUnlock()
+	var gvks []schema.GroupVersionKind
+	for k := range r.intent {
+		for k2 := range r.intent[k] {
+			gvks = append(gvks, k2)
```

Comment:
This has the potential of double-counting GVKs, as more than one registrar could be listing a watch a single kind.

Part of the cleanup I was going to do was moving the registrar merge logic into the recordKeeper. I suggest:

```
func (r *recordKeeper) Get() map[schema.GroupVersionKind]vitals {
	r.intentMux.RLock()
	defer r.intentMux.RUnlock()
	cpy := make(map[string]map[schema.GroupVersionKind]vitals)
	for k := range r.intent {
		cpy[k] = make(map[schema.GroupVersionKind]vitals)
		for k2, v := range r.intent[k] {
			cpy[k][k2] = v
		}
	}
	managedKinds := make(map[schema.GroupVersionKind]vitals)
	for _, registrar := range cpy {
		for gvk, v := range registrar {
			if mk, ok := managedKinds[gvk]; ok {
				merged, err := mk.merge(v)
				if err != nil {
					return nil, nil, nil, err
				}
				managedKinds[gvk] = merged
			} else {
				managedKinds[gvk] = v
			}
		}
	}
        return managedKinds
}
```

Then GetGVK becomes:

```
func (r *recordKeeper) GetGVK() []schema.GroupVersionKind {
	var gvks []schema.GroupVersionKind
	for gvk := range r.Get() {
	   gvks = append(gvks, gvk)
       }
       return gvks
```

Note that a lock is not needed because the Get() function uses a read lock and returns a copy of the data, so this function is thread-safe without it.